packages/get-credentials/src/get-credentials.js

```diff
@@ -0,0 +1,60 @@
+import dotenv from 'dotenv'
+import path from 'path'
+
+import { homepage } from '../package.json'
+
+export default function getCredentials (projectKey) {
+  if (!projectKey)
+    return Promise.reject([new Error('Project Key is needed')])
+
+  return Promise.resolve(setCredentialsFromEnvFile())
+    .catch(Promise.resolve)
+    .then(dotenvError => getCredentialsFromEnvironment(projectKey)
+      .catch(environmentError =>
+        Promise.reject([dotenvError, environmentError]),
+      ))
```

Comment:
Yes since multiple things can go wrong but don't want to immediately stop and return an error when they do it rejects at the end 'collecting' the errors before that into an array.